### PR TITLE
Fix NFT allowance acceptance tests

### LINKED ISSUE #8270: Enable NFT allowance @acceptance tag in  tests to remove conflict with ContractCall tests
### Problem

Since the NFT allowance APIs use the same token as the contract call API to test , there was an issue after deleting the token.

To reproduce:
1. Add @acceptance in the nftAllowance.feature and run the  acceptance tests on integration using a custom image.

### Solution

- We need to check how the serial numbers for the token are being set. After deletion that map should still be available to any other tests that use the same token.

Another line item is to change documentation in the openapi spec.
### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by mgoelswirlds**
**Description**:
This PR allows the acceptance tests for NFT allowances to run under the @acceptance tag.

This PR modifies
* Fixing regex for Token,Account and PauseStatus Enum validation
* NFT's state to allow sharing between classes.
* Adding @acceptance tag to the NftAllowance feature
* Changing openapi text to remove the experimental clause.

**Related issue(s)**:

Fixes #8270

**Notes for reviewer**:
There are existing open PRs which would need to be merged to add all NFT allowances fnctionality.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by jascks — DISMISSED** [review]
LGTM

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TokenClient.java [hunk 1]
```diff
@@ -628,6 +628,11 @@ public enum TokenNameEnum {
                 TokenKycStatus.KycNotApplicable,
                 TokenFreezeStatus.FreezeNotApplicable),
         NFT("non_fungible", TokenType.NON_FUNGIBLE_UNIQUE, TokenKycStatus.KycNotApplicable, TokenFreezeStatus.Unfrozen),
+        ALLOWANCENONFUNGIBLE(
+                "allowance_non_fungible",
+                TokenType.NON_FUNGIBLE_UNIQUE,
+                TokenKycStatus.KycNotApplicable,
+                TokenFreezeStatus.FreezeNotApplicable),
```
Comment (jascks) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TokenClient.java:635]: After you merge in the `main` branch, you will find that this enum type added `boolean createWithMetadata`. Go ahead and set that to `false` like the great majority of the tokens.

Comment (mgoelswirlds) [reply hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TokenClient.java:635]: Thanks for that update. FIxing it now.

**Review by jascks — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Comment by mgoelswirlds** [comment]
@XIn found a fix to get around creating a new token to fix state. Added that now.

**Review by jascks — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM. don't forget to update the description since we put it the squashed merge message

**Review by jnels124 — APPROVED** [review]
LGTM

**Pull request merged**